engine/execution/computation/programs_test.go

```diff
@@ -0,0 +1,368 @@
+
+	})
+	t.Run("executing block1121 (emit event (expected v4))", func(t *testing.T) {
+		block1121ExpectedValue := 4
+		block1121tx1 := testutil.CreateEmitEventTransaction(account, account)
+		prepareTx(t, block1121tx1, account, privKey, 3, chain)
+
+		col1121 := flow.Collection{Transactions: []*flow.TransactionBody{block1121tx1}}
+		block1121View = block112View.NewChild()
+		block1121, res = createTestBlockAndRun(t, engine, block112, col1121, block1121View)
+		// cache should include a program for this block
+		require.NotNil(t, programsCache.Get(block1121.ID()))
+		// 1st event
+		hasValidEventValue(t, res.Events[0], block1121ExpectedValue)
+
+	})
+	t.Run("executing block12 (deploys contract V2)", func(t *testing.T) {
+
+		block12tx1 := testutil.DeployEventContractTransaction(account, chain, 2)
+		prepareTx(t, block12tx1, account, privKey, 0, chain)
+
+		col12 := flow.Collection{Transactions: []*flow.TransactionBody{block12tx1}}
+		block12View = block1View.NewChild()
+		block12, res = createTestBlockAndRun(t, engine, block1, col12, block12View)
+		// cache should include a program for this block
+		require.NotNil(t, programsCache.Get(block12.ID()))
+		assert.EqualValues(t, "flow.AccountContractAdded", res.Events[0].Type)
+	})
+	t.Run("executing block121 (emit event (expected V2)", func(t *testing.T) {
+		block121ExpectedValue := 2
+		block121tx1 := testutil.CreateEmitEventTransaction(account, account)
+		prepareTx(t, block121tx1, account, privKey, 1, chain)
+
+		col121 := flow.Collection{Transactions: []*flow.TransactionBody{block121tx1}}
+		block121View = block12View.NewChild()
+		block121, res = createTestBlockAndRun(t, engine, block12, col121, block121View)
+		// cache should include a program for this block
+		require.NotNil(t, programsCache.Get(block121.ID()))
+		// 1st event
+		hasValidEventValue(t, res.Events[0], block121ExpectedValue)
+	})
+	t.Run("executing Block1211 (emit event (expected V2)", func(t *testing.T) {
+		block1211ExpectedValue := 2
+		block1211tx1 := testutil.CreateEmitEventTransaction(account, account)
+		prepareTx(t, block1211tx1, account, privKey, 2, chain)
+
+		col1211 := flow.Collection{Transactions: []*flow.TransactionBody{block1211tx1}}
+		block1211View = block121View.NewChild()
+		block1211, res = createTestBlockAndRun(t, engine, block121, col1211, block1211View)
+		// cache should include a program for this block
+		require.NotNil(t, programsCache.Get(block1211.ID()))
+		// had no change so cache should be equal to parent
+		require.Equal(t, programsCache.Get(block111.ID()), programsCache.Get(block1111.ID()))
+		// 1st event
+		hasValidEventValue(t, res.Events[0], block1211ExpectedValue)
+	})
+
+}
+
+func createTestBlockAndRun(t *testing.T, engine *Manager, parentBlock *flow.Block, col flow.Collection, view *delta.View) (*flow.Block, *execution.ComputationResult) {
+	guarantee := flow.CollectionGuarantee{
+		CollectionID: col.ID(),
+		Signature:    nil,
+	}
+
+	block := &flow.Block{
```

Comment:
very minor, but we have fixtures to create blocks with parents.

Reply:
I would like to keep it this way, atleast for now, I want to expand this test for some other fvm stuff as well and prefer not to update the original fixture.